src/Command/PurgeCommand.php

```diff
@@ -0,0 +1,318 @@
+    private function purgeLeads(array $leads, array $masterForm)
+    {
+        $deleted = 0;
+        $ids = implode(',', array_keys($leads));
+
+        if (!empty($ids)) {
+            $deleted = $this->db->executeUpdate(
+                "DELETE FROM tl_lead WHERE id IN(".$ids.")"
+            );
+
+            $logLevel = LogLevel::INFO;
+            $logMessage = 'Purged '.(int)$deleted.' leads for master form "'.$masterForm['title'].'"';
+            $this->logger->log($logLevel, $logMessage, array('contao' => new ContaoContext(__METHOD__, $logLevel)));
+        }
+
+        return $deleted;
+    }
+
+    /**
+     * @param array $leads
+     * @return array
+     */
+    private function getAllLeadsData(array $leads)
+    {
+        $leadsData = [];
+
+        $ids = implode(',', array_keys($leads));
+
+        if (!empty($ids)) {
+
+            $rows = $this->db->fetchAll(
+                "SELECT d.*, f.type AS field_type FROM tl_lead_data d
+                      LEFT JOIN tl_form_field f ON d.field_id = f.id
+                      WHERE d.pid IN(".$ids.")"
+            );
+
+            foreach ($rows as $row) {
+                $leadsData[$row['id']] = $row;
+            }
+        }
+
+
+        return $leadsData;
+    }
+
+    /**
+     * @param array $leadsData
+     * @param array $masterForm
+     * @return int
+     * @throws \Doctrine\DBAL\DBALException
+     */
+    private function purgeLeadsData(array $leadsData, array $masterForm)
+    {
+        $deleted = 0;
+        $ids = implode(',', array_keys($leadsData));
+
+        if (!empty($ids)) {
+            $deleted = $this->db->executeUpdate(
+                "DELETE FROM tl_lead_data WHERE id IN(".$ids.")"
+            );
+
+            $logLevel = LogLevel::INFO;
+            $logMessage = 'Purged '.(int)$deleted.' leads data for master form "'.$masterForm['title'].'"';
+            $this->logger->log($logLevel, $logMessage, array('contao' => new ContaoContext(__METHOD__, $logLevel)));
+        }
+
+        return $deleted;
+    }
+
+    /**
+     * @param array $leadsData
+     * @return array
+     */
+    private function purgeUploads(array $leadsData)
+    {
+        $files = [];
+        if (!empty($leadsData)) {
+            foreach ($leadsData as $data) {
+                if ('upload' === $data['field_type']) {
+                    $files[$data['id']] = $this->purgeUpload($data['value']);
+                }
+            }
+        }
+
+        return $files;
+    }
+
+    /**
+     * @param string $value
+     * @return FilesModel|null
+     */
+    private function purgeUpload(string $value)
+    {
+        if (!Validator::isUuid($value)) {
+            return null;
+        }
+
+        $logLevel = LogLevel::INFO;
+        $logMessage = 'Purge leads upload "'.$value.'": ';
+
+        $filesModel = FilesModel::findByUUid($value);
+
+        if (null !== $filesModel) {
+            try {
+                if ($this->fs->exists(TL_ROOT.'/'.$filesModel->path)) {
```

Comment:
Instead of using an old constant like `TL_ROOT` you can inject `kernel.project_dir`. 😄

Reply:
I'm to used to 'tüde lü' ;-)